clean.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b         Delete bin directory"
+    echo "  -p         Delete packages directory"
+    echo "  -t         Delete Tools directory"
+    echo "  -w         Remove all untracked files in the working tree, including the bin, packages, and Tools directories"
+    echo "  -c         Delete the user-local NuGet package caches, located in ~/.local/share/dnu and ~/.local/share/NuGet"
+    echo "  -a, --all  Clean all of the above"
+    exit 1
+}
+
+invoke_buildproj()
+{
+    PackageRoot=$WorkingTreeRoot/packages
```

Comment:
This is going to be changing when it uses the .NET Core MSBuild instead of Mono, which is dependent on changes Joe is working on.